environments/builder/builder.go

```diff
@@ -0,0 +1,154 @@
+package builder
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/mholt/archiver"
+
+	"github.com/fission/fission"
+)
+
+const (
+	pathPrefix      = "/tmp/fission"
+	srcPkgPath      = pathPrefix + "/srcPkg"
+	deployPkgPath   = pathPrefix + "/deployPkg"
+	archiveFilename = "archive.zip"
+
+	// supported environment variables
+	envSrcPkg    = "SRC_PKG"
+	envDeployPkg = "DEPLOY_PKG"
+)
+
+type (
+	Builder struct {
+		sharedVolumePath string
+	}
+)
+
+func MakeBuilder(sharedVolumePath string) *Builder {
+	return &Builder{
+		sharedVolumePath: sharedVolumePath,
+	}
+}
+
+func (builder *Builder) Handler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, "", 404)
+		return
+	}
+
+	startTime := time.Now()
+	defer func() {
+		elapsed := time.Now().Sub(startTime)
+		log.Printf("elapsed time in build request = %v", elapsed)
+	}()
+
+	// parse request
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Error reading request body")
+		http.Error(w, err.Error(), 500)
+		return
+	}
+	var req fission.PackageBuildRequest
+	err = json.Unmarshal(body, &req)
+	if err != nil {
+		log.Printf("Error reading request body: %v", err)
+		http.Error(w, err.Error(), 500)
+		return
+	}
+	log.Printf("builder received request: %v", req)
+
+	srcPkgArchivePath := filepath.Join(builder.sharedVolumePath, req.SrcPkgFilename)
```

Comment:
can you add some log messages here?  "Uncompressing package", "starting build", etc.